dao/src/main/java/greencity/repository/PlaceRepo.java

```diff
@@ -79,4 +80,13 @@ List<Place> findAllByModifiedDateBetweenAndStatus(ZonedDateTime startDate, Zoned
             + "WHERE c.name IN (:category) "
             + "or c.name_ua IN (:category)")
     List<Place> findPlaceByCategory(String[] category);
+
+    /**
+     * Finds a place by its name.
+     *
+     * @param name the name of the place.
+     * @return an optional containing the place if found, or empty otherwise.
+     */
+    @Query("SELECT p FROM Place p WHERE LOWER(p.name) = LOWER(:name)")
+    Optional<Place> findByNameIgnoreCase(@Param("name") String name);
```

Comment:
is mane a unique field?
if not we will have a list here

Reply:
In general, we don't have any annotations on the placeName field to guarantee uniqueness, nor is there a uniqueness check in the service's creation method. However, the task was described as if the name of each place is unique. The task also specified that we accept the following JSON structure:
{
  "placeName": "",
  "newStatus": "",
  "userName": "",
  "email": ""
}
Since there’s no unique field to rely on, I implemented it this way.